### PR TITLE
Bahrain holidays

### PR DESCRIPTION
**Pull request by taljishi**
Adding support for Bahrain holidays bahrain.py.

This is my first contribution, and since I'm not a programmer or coder, I based some of the code with Saudi Arabia and UAE to get a head start

### CODE REVIEW

File: holidays/countries/bahrain.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        # Ashoora
+        # Date is announced each year, for the 9th and 10th Day,
+        # of the month of Muharam
+        ashoora = "Ashoora"
+        if year in dates_obs:
+            for date_obs in dates_obs[year]:
+                hol_date = date(year, *date_obs)
+                self[hol_date + rd(days=9)] = ashoora
+                self[hol_date + rd(days=10)] = f"{ashoora} Holiday"
```
Comment (KJhellico) [inline holidays/countries/bahrain.py:147]: If Ashoora calculated as 9 Muharram, it's no need to add 9
```suggestion
                self[hol_date] = ashoora
                self[hol_date + rd(days=+1)] = f"{ashoora} Holiday"
```

File: holidays/countries/bahrain.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        # National Day
+        self[date(year, DEC, 16)] = "National Day"
+        self[date(year, DEC, 17)] = "National Day"
+
+        # Eid Al Fitr
+        # Date is announced each year. Usually stretches along 3 or 4 days,
+        # in some instances prepending/appending a day or two
+        # before/after the official holiday.
+        fitr = "Eid Al Fitr"
+        if year in dates_obs:
```
Comment (KJhellico) [inline holidays/countries/bahrain.py:81]: In that case, you should define some known dates list as `dates_obs`

**Comment by arkid15r** [comment]
@KJhellico, yeah, this PR's code is somewhat incomplete but that's fine. It's a draft so I believe no review was required. I'm going to take over @taljishi's work from here as we agreed in https://github.com/dr-prodigy/python-holidays/discussions/813

File: holidays/countries/bahrain.py (hunk 2 above)
Comment (arkid15r) [reply holidays/countries/bahrain.py:81]: I'll try to find out the correct values for date objects soon.

**Pull request closed without merging**